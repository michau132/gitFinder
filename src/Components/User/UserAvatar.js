import React from 'react';
import styled from 'styled-components';
import { observer, PropTypes, inject } from 'mobx-react';

const Image = styled.img`
  object-fit: contain;
  max-width: 100%;
  max-height: 100%;
`;

const UserAvatar = ({ store: { informations: { avatar_url } } }) => (
  <Image src={avatar_url} alt="avatar" />
);

UserAvatar.propTypes = {
  store: PropTypes.observableObject.isRequired,
};

export default inject('store')(observer(UserAvatar));
