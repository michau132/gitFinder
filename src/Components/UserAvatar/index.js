import React from 'react';
import PropTypes from 'prop-types';
import styles from './style.css';

const UserAvatar = ({ avatar }) => {
  return (
    <div className={styles.userAvatar}>
      <img src={avatar} alt="avatar" className={styles.avatar} />
    </div>
  );
};

UserAvatar.propTypes = {
  
};

export default UserAvatar;