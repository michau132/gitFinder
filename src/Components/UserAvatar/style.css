.userAvatar {
  overflow: hidden;
}

.avatar {
  max-width: 100%;
  max-height: 100%;
}