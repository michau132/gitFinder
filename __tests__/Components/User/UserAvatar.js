import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserAvatar from '../../../src/Components/User/UserAvatar';

describe('testing UserAvatar component', () => {
  test('matching snapshot', () => {
    const wrapper = shallow(<UserAvatar avatarUrl="https://via.placeholder.com/150" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
