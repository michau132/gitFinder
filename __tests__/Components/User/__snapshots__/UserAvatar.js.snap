// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing UserAvatar component matching snapshot 1`] = `
<styled.img
  alt="avatar"
  src="https://via.placeholder.com/150"
/>
`;
